Services - Subscriber only section

noruzzamans opened the pull request:
Testing Instructions
Create a new page.
Add the pattern.
Confirm that it looks like the designs (spacing/colors/etc.).

Closes https://github.com/WordPress/twentytwentyfive/issues/77
[Figma](https://www.figma.com/design/dzGCSntVch4EQdVERTqyVK/Twenty-Twenty-Five?node-id=3177-1808&node-type=SYMBOL&t=PTgblFBFp3QdQSCt-0)

beafialho commented:
Thank you for working on this one @noruzzamans! Great job.

I noticed a few things that should be changed to match the design, such as:
- You used two different blocks for buttons, this isn't necessary, you can add both in the same Buttons block
- You used a columns block inside a columns block, which also seems unnecessary
- The "become a member" sentence shouldn't be a button and wasn't looking like the "pill" style (this one is an open discussion but it most definitely isn't a button, like in your markup)
- There were a few spelling mistakes, and some letters that should be capitals
- Spacing is a bit different than in Figma
- All the things I mentioned above also make the pattern not looking good on mobile

Here's a suggestion with these adjustments, so you can compare with your current pattern.

```
<!-- wp:group {"metadata":{"categories":["call-to-action","featured","services"],"patternName":"twentytwentyfive/services-subscriber-only-section","name":"Services - Subscriber only section"},"align":"full","style":{"spacing":{"blockGap":"var:preset|spacing|50","padding":{"top":"var:preset|spacing|70","bottom":"var:preset|spacing|70"}}},"layout":{"type":"constrained"}} -->
<div class="wp-block-group alignfull" style="padding-top:var(--wp--preset--spacing--70);padding-bottom:var(--wp--preset--spacing--70)"><!-- wp:columns {"align":"wide","style":{"spacing":{"blockGap":{"top":"var:preset|spacing|70","left":"var:preset|spacing|70"}}}} -->
<div class="wp-block-columns alignwide"><!-- wp:column {"verticalAlignment":"center"} -->
<div class="wp-block-column is-vertically-aligned-center"><!-- wp:heading {"style":{"typography":{"fontStyle":"normal","fontWeight":"700"}},"fontSize":"small"} -->
<h2 class="wp-block-heading has-small-font-size" style="font-style:normal;font-weight:700">Become a member</h2>
<!-- /wp:heading -->

<!-- wp:heading {"level":3,"fontSize":"xx-large"} -->
<h3 class="wp-block-heading has-xx-large-font-size">Subscribe to get unlimited access</h3>
<!-- /wp:heading -->

<!-- wp:list {"style":{"spacing":{"margin":{"top":"var:preset|spacing|30","bottom":"var:preset|spacing|30"}}}} -->
<ul style="margin-top:var(--wp--preset--spacing--30);margin-bottom:var(--wp--preset--spacing--30)" class="wp-block-list"><!-- wp:list-item {"fontSize":"medium"} -->
<li class="has-medium-font-size">Get access to our paid articles and weekly newsletter.</li>
<!-- /wp:list-item -->

<!-- wp:list-item {"fontSize":"medium"} -->
<li class="has-medium-font-size">Join our IRL event.</li>
<!-- /wp:list-item -->

<!-- wp:list-item {"fontSize":"medium"} -->
<li class="has-medium-font-size">Get a free tote bag.</li>
<!-- /wp:list-item -->

<!-- wp:list-item {"fontSize":"medium"} -->
<li class="has-medium-font-size">An elegant addition of home decor collection.</li>
<!-- /wp:list-item -->

<!-- wp:list-item {"fontSize":"medium"} -->
<li class="has-medium-font-size">Join our forums.</li>
<!-- /wp:list-item --></ul>
<!-- /wp:list -->

<!-- wp:buttons {"layout":{"type":"flex","justifyContent":"left","flexWrap":"nowrap"}} -->
<div class="wp-block-buttons"><!-- wp:button {"className":"is-style-fill"} -->
<div class="wp-block-button is-style-fill"><a class="wp-block-button__link wp-element-button">Subscribe</a></div>
<!-- /wp:button -->

<!-- wp:button {"className":"is-style-outline"} -->
<div class="wp-block-button is-style-outline"><a class="wp-block-button__link wp-element-button">View plans</a></div>
<!-- /wp:button --></div>
<!-- /wp:buttons -->

<!-- wp:paragraph {"fontSize":"small"} -->
<p class="has-small-font-size">Cancel or pause anytime.</p>
<!-- /wp:paragraph --></div>
<!-- /wp:column -->

<!-- wp:column {"verticalAlignment":"center"} -->
<div class="wp-block-column is-vertically-aligned-center"><!-- wp:image {"id":76,"sizeSlug":"full","linkDestination":"none"} -->
<figure class="wp-block-image size-full"><img src="http://tt5.local/wp-content/themes/twentytwentyfive/assets/images/services-subscriber-photo.webp" alt="Smartphones capturing a scenic wildflower meadow with trees" class="wp-image-76"/></figure>
<!-- /wp:image --></div>
<!-- /wp:column --></div>
<!-- /wp:columns --></div>
<!-- /wp:group -->
```

noruzzamans commented:
> Thank you for working on this one @noruzzamans! Great job.
> 
> I noticed a few things that should be changed to match the design, such as:
> 
> * You used two different blocks for buttons, this isn't necessary, you can add both in the same Buttons block
> * You used a columns block inside a columns block, which also seems unnecessary
> * The "become a member" sentence shouldn't be a button and wasn't looking like the "pill" style (this one is an open discussion but it most definitely isn't a button, like in your markup)
> * There were a few spelling mistakes, and some letters that should be capitals
> * Spacing is a bit different than in Figma
> * All the things I mentioned above also make the pattern not looking good on mobile
> 
> Here's a suggestion with these adjustments, so you can compare with your current pattern.
> 
> ```
> <!-- wp:group {"metadata":{"categories":["call-to-action","featured","services"],"patternName":"twentytwentyfive/services-subscriber-only-section","name":"Services - Subscriber only section"},"align":"full","style":{"spacing":{"blockGap":"var:preset|spacing|50","padding":{"top":"var:preset|spacing|70","bottom":"var:preset|spacing|70"}}},"layout":{"type":"constrained"}} -->
> <div class="wp-block-group alignfull" style="padding-top:var(--wp--preset--spacing--70);padding-bottom:var(--wp--preset--spacing--70)"><!-- wp:columns {"align":"wide","style":{"spacing":{"blockGap":{"top":"var:preset|spacing|70","left":"var:preset|spacing|70"}}}} -->
> <div class="wp-block-columns alignwide"><!-- wp:column {"verticalAlignment":"center"} -->
> <div class="wp-block-column is-vertically-aligned-center"><!-- wp:heading {"style":{"typography":{"fontStyle":"normal","fontWeight":"700"}},"fontSize":"small"} -->
> <h2 class="wp-block-heading has-small-font-size" style="font-style:normal;font-weight:700">Become a member</h2>
> <!-- /wp:heading -->
> 
> <!-- wp:heading {"level":3,"fontSize":"xx-large"} -->
> <h3 class="wp-block-heading has-xx-large-font-size">Subscribe to get unlimited access</h3>
> <!-- /wp:heading -->
> 
> <!-- wp:list {"style":{"spacing":{"margin":{"top":"var:preset|spacing|30","bottom":"var:preset|spacing|30"}}}} -->
> <ul style="margin-top:var(--wp--preset--spacing--30);margin-bottom:var(--wp--preset--spacing--30)" class="wp-block-list"><!-- wp:list-item {"fontSize":"medium"} -->
> <li class="has-medium-font-size">Get access to our paid articles and weekly newsletter.</li>
> <!-- /wp:list-item -->
> 
> <!-- wp:list-item {"fontSize":"medium"} -->
> <li class="has-medium-font-size">Join our IRL event.</li>
> <!-- /wp:list-item -->
> 
> <!-- wp:list-item {"fontSize":"medium"} -->
> <li class="has-medium-font-size">Get a free tote bag.</li>
> <!-- /wp:list-item -->
> 
> <!-- wp:list-item {"fontSize":"medium"} -->
> <li class="has-medium-font-size">An elegant addition of home decor collection.</li>
> <!-- /wp:list-item -->
> 
> <!-- wp:list-item {"fontSize":"medium"} -->
> <li class="has-medium-font-size">Join our forums.</li>
> <!-- /wp:list-item --></ul>
> <!-- /wp:list -->
> 
> <!-- wp:buttons {"layout":{"type":"flex","justifyContent":"left","flexWrap":"nowrap"}} -->
> <div class="wp-block-buttons"><!-- wp:button {"className":"is-style-fill"} -->
> <div class="wp-block-button is-style-fill"><a class="wp-block-button__link wp-element-button">Subscribe</a></div>
> <!-- /wp:button -->
> 
> <!-- wp:button {"className":"is-style-outline"} -->
> <div class="wp-block-button is-style-outline"><a class="wp-block-button__link wp-element-button">View plans</a></div>
> <!-- /wp:button --></div>
> <!-- /wp:buttons -->
> 
> <!-- wp:paragraph {"fontSize":"small"} -->
> <p class="has-small-font-size">Cancel or pause anytime.</p>
> <!-- /wp:paragraph --></div>
> <!-- /wp:column -->
> 
> <!-- wp:column {"verticalAlignment":"center"} -->
> <div class="wp-block-column is-vertically-aligned-center"><!-- wp:image {"id":76,"sizeSlug":"full","linkDestination":"none"} -->
> <figure class="wp-block-image size-full"><img src="http://tt5.local/wp-content/themes/twentytwentyfive/assets/images/services-subscriber-photo.webp" alt="Smartphones capturing a scenic wildflower meadow with trees" class="wp-image-76"/></figure>
> <!-- /wp:image --></div>
> <!-- /wp:column --></div>
> <!-- /wp:columns --></div>
> <!-- /wp:group -->
> ```

Thanks for your review. I have followed your guidelines and updated the pattern. Please review it and let me know if any changes are needed.

beafialho commented on patterns/services-subscriber-only-section.php:
```diff
@@ -0,0 +1,68 @@
+<div class="wp-block-group alignfull" style="padding-top:var(--wp--preset--spacing--70);padding-bottom:var(--wp--preset--spacing--70)"><!-- wp:columns {"align":"wide","style":{"spacing":{"blockGap":{"top":"var:preset|spacing|70","left":"var:preset|spacing|70"}}}} -->
+<div class="wp-block-columns alignwide"><!-- wp:column {"verticalAlignment":"center"} -->
+<div class="wp-block-column is-vertically-aligned-center"><!-- wp:heading {"style":{"typography":{"fontStyle":"normal","fontWeight":"700"}},"fontSize":"small"} -->
+<h2 class="wp-block-heading has-small-font-size" style="font-style:normal;font-weight:700">Become a member</h2>
```
Let's just remove the "Become a member" heading. As per the discussion in #150.

carolinan reviewed (changes requested):
Hi!
I'd like to make the following change requests:

- Please remove the unrelated pattern file: services-three-column-boxed-pricing-table.php.
- Leave an empty blank line at the end of services-subscriber-only-section.php.
- Apply the checkmark block style to the list.
- Increase the spacing between the list and the buttons, to match the design.
- Add the license and source information for the image in the readme.txt file.

noruzzamans commented:
> Hi! I'd like to make the following change requests:
> 
> * Please remove the unrelated pattern file: services-three-column-boxed-pricing-table.php.
> * Leave an empty blank line at the end of services-subscriber-only-section.php.
> * Apply the checkmark block style to the list.
> * Increase the spacing between the list and the buttons, to match the design.
> * Add the license and source information for the image in the readme.txt file.

@carolinan I have removed the unrelated pattern file, left an empty line, and added the checkmark block style to the list. I also increased the spacing between the list and buttons to match the design. I have seen the image source information already listed in the README file. If this is not the correct image source, could you please let me know how to find the right link? Thanks

beafialho commented:
Thank you @noruzzamans. Somehow I'm still seeing the "Become a member" heading, which we can remove. I'm not seeing the recent changes requested by @carolinan. Could you add a screenshot/recording?

noruzzamans commented:
> Thank you @noruzzamans. Somehow I'm still seeing the "Become a member" heading, which we can remove. I'm not seeing the recent changes requested by @carolinan. Could you add a screenshot/recording?

@beafialho Could you please clarify whether you want to completely remove the 'Become a member' section or update its heading to a pill button style?

noruzzamans commented:
> > Thank you @noruzzamans. Somehow I'm still seeing the "Become a member" heading, which we can remove. I'm not seeing the recent changes requested by @carolinan. Could you add a screenshot/recording?
> 
> @beafialho Could you please clarify whether you want to completely remove the 'Become a member' section or update its heading to a pill button style?

@beafialho Apologies for the previous comment. I had previously seen the 'Become a member' section in the design. However, after rechecking the Figma file, I noticed it has been removed. I have now removed it and updated the pattern. Please review and let me know. Many thanks for your cooperation.

carolinan commented:
I think the only remaining change is adding the image license and credits to the readme.txt.
I am guessing @beafialho created this device mockup, using the same image that is used in the coming soon pattern, coming-soon-bg-image.webp ?
Then the readme should mention that the license is public domain and that mockup is created by her with this image.

beafialho commented:
> I am guessing @beafialho created this device mockup, using the same image that is used in the coming soon pattern, coming-soon-bg-image.webp ?
> Then the readme should mention that the license is public domain and that mockup is created by her with this image.

That is correct 👍

juanfra commented on patterns/services-subscriber-only-section.php:
```diff
@@ -0,0 +1,64 @@
+<!-- /wp:heading -->
+
+<!-- wp:list {"className":"is-style-checkmark-list","style":{"spacing":{"margin":{"top":"var:preset|spacing|40","bottom":"var:preset|spacing|40"}}}} -->
+<ul style="margin-top:var(--wp--preset--spacing--40);margin-bottom:var(--wp--preset--spacing--40)" class="wp-block-list is-style-checkmark-list"><!-- wp:list-item {"fontSize":"medium"} -->
```
It looks like we can set the padding-left of this list to x-small to make it match the design.

noruzzamans replied:
@juanfra Thanks for reviewing. I have set the padding-left of this list to x-small. Please take a look and let me know your thoughts.

juanfra commented on patterns/services-subscriber-only-section.php:
```diff
@@ -0,0 +1,64 @@
+<!-- /wp:column -->
+
+<!-- wp:column {"verticalAlignment":"center"} -->
+<div class="wp-block-column is-vertically-aligned-center"><!-- wp:image {"id":76,"sizeSlug":"full","linkDestination":"none"} -->
```
```suggestion
<div class="wp-block-column is-vertically-aligned-center"><!-- wp:image {"sizeSlug":"full","linkDestination":"none"} -->
```

noruzzamans replied:
> 



> 

@juanfra I have removed this "id":76,

juanfra commented on patterns/services-subscriber-only-section.php:
```diff
@@ -0,0 +1,64 @@
+
+<!-- wp:column {"verticalAlignment":"center"} -->
+<div class="wp-block-column is-vertically-aligned-center"><!-- wp:image {"id":76,"sizeSlug":"full","linkDestination":"none"} -->
+<figure class="wp-block-image size-full"><img src="<?php echo esc_url( get_template_directory_uri() ); ?>/assets/images/services-subscriber-photo.webp" alt="Smartphones capturing a scenic wildflower meadow with trees" class="wp-image-76"/></figure>
```
```suggestion
<figure class="wp-block-image size-full"><img src="<?php echo esc_url( get_template_directory_uri() ); ?>/assets/images/services-subscriber-photo.webp" alt="Smartphones capturing a scenic wildflower meadow with trees" /></figure>
```

noruzzamans replied:
> 

@juanfra I have removed this class="wp-image-76"

juanfra reviewed (commented):
Thanks for working on this. I left a few comments.

It also would be nice if the code could be with the right indent. Thank you!

juanfra commented on patterns/services-subscriber-only-section.php:
```diff
@@ -0,0 +1,64 @@
+<div class="wp-block-group alignfull" style="padding-top:var(--wp--preset--spacing--70);padding-bottom:var(--wp--preset--spacing--70)"><!-- wp:columns {"align":"wide","style":{"spacing":{"blockGap":{"top":"var:preset|spacing|70","left":"var:preset|spacing|70"}}}} -->
+<div class="wp-block-columns alignwide"><!-- wp:column {"verticalAlignment":"center"} -->
+<div class="wp-block-column is-vertically-aligned-center"><!-- wp:heading {"fontSize":"xx-large"} -->
+<h2 class="wp-block-heading has-xx-large-font-size">Subscribe to get unlimited access</h2>
```
This will also don't match the size of the design, but we don't have any other alternative now. Leaving this comment to come back to it later when we have proper sizes.

noruzzamans replied:
@juanfra Okay, I haven't updated this part yet. Once we get the proper sizes, I'll update it.

juanfra replied:
Thanks! Ok, we can move forward with this as is and revisit later.

juanfra commented on patterns/services-subscriber-only-section.php:
```diff
@@ -0,0 +1,64 @@
+ */
+
+?>
+<!-- wp:group {"align":"full","style":{"spacing":{"blockGap":"var:preset|spacing|50","padding":{"top":"var:preset|spacing|70","bottom":"var:preset|spacing|70"}}},"layout":{"type":"constrained"}} -->
```
```suggestion
<!-- wp:group {"align":"full","style":{"spacing":{"blockGap":"var:preset|spacing|50","padding":{"top":"var:preset|spacing|70","bottom":"var:preset|spacing|70"},"margin":{"top":"0","bottom":"0"}}},"layout":{"type":"constrained"}} -->
```

juanfra commented on patterns/services-subscriber-only-section.php:
```diff
@@ -0,0 +1,64 @@
+
+?>
+<!-- wp:group {"align":"full","style":{"spacing":{"blockGap":"var:preset|spacing|50","padding":{"top":"var:preset|spacing|70","bottom":"var:preset|spacing|70"}}},"layout":{"type":"constrained"}} -->
+<div class="wp-block-group alignfull" style="padding-top:var(--wp--preset--spacing--70);padding-bottom:var(--wp--preset--spacing--70)"><!-- wp:columns {"align":"wide","style":{"spacing":{"blockGap":{"top":"var:preset|spacing|70","left":"var:preset|spacing|70"}}}} -->
```
```suggestion
<div class="wp-block-group alignfull" style="margin-top:0;margin-bottom:0;padding-top:var(--wp--preset--spacing--70);padding-bottom:var(--wp--preset--spacing--70)"><!-- wp:columns {"align":"wide","style":{"spacing":{"blockGap":{"top":"var:preset|spacing|70","left":"var:preset|spacing|70"}}}} -->
```

juanfra reviewed (approved):
Thank you for working on this @noruzzamans 🚀
